cmake_modules/FindCRYPTOLENS.cmake

```diff
@@ -0,0 +1,64 @@
+#  Copyright (C) 2015 3D Repo Ltd
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero General Public License as
+#  published by the Free Software Foundation, either version 3 of the
+#  License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Affero General Public License for more details.
+#
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+# attempt to find CRYPTOLENS C++ driver 
+# if CRYPTOLENS is found, CRYPTOLENS_FOUND is set to true
+# CRYPTOLENS_INCLUDE_DIR will point to the include folder of the installation
+# CRYPTOLENS_LIBRARIES will point to the libraries
+#
+# Assuming the folder structure is:
+# cryptolens_root
+#   - include
+#   - lib
+#     - Debug
+#     - Release
+#
+# start the search by looking for the lib in the 
+# root DIR of of CRYPTOLENS
+if(DEFINED ENV{CRYPTOLENS_ROOT})
+	set(CRYPTOLENS_ROOT $ENV{CRYPTOLENS_ROOT})
+	message(STATUS "$CRYPTOLENS_ROOT defined: ${CRYPTOLENS_ROOT}")
+	find_path(CRYPTOLENS_INCLUDE_DIR cryptolens/core.hpp
+		${CRYPTOLENS_ROOT}/include
+		)    
+	find_library(CRYPTOLENS_LIBRARIES_RELEASE 
+		NAMES cryptolens
+		PATHS
+		${CRYPTOLENS_ROOT}/lib/Release
+	)	
+	find_library(CRYPTOLENS_LIBRARIES_DEBUG 
+		NAMES cryptolens
+		PATHS
+		${CRYPTOLENS_ROOT}/lib/Debug
+	)
+	set(CRYPTOLENS_LIBRARIES
+		debug ${CRYPTOLENS_LIBRARIES_DEBUG}
+		optimized ${CRYPTOLENS_LIBRARIES_RELEASE}
+		)
+endif()
+
```

Comment:
Can you copy over the auto-detection parts as well from one of the other scripts?